dash/App/Accounts/Add/AddAddress/index.js

```diff
@@ -54,17 +48,30 @@ class AddAddress extends React.Component {
 
   create () {
     this.setState({ index: ++this.state.index })
-    link.rpc('createFromAddress', this.state.address, 'Watch Account', (err) => {
+    const {address: input} = this.state
+    const createAccountCallback =  (address) => link.rpc('createFromAddress', address, 'Watch Account', (err) => {
       if (err) {
         this.setState({ status: err, error: true })
       } else {
         this.setState({ status: 'Successful', error: false })
       }
     })
+
+    if( input.toLowerCase().includes('.eth') ) {
+      link.rpc('resolveEnsName', input, (err, resolvedAddress) => {
+        if(err){
+          this.setState({ status: `Unable to resolve Ethereum address for ${input}`, error: true })
+        } else {
+          createAccountCallback(resolvedAddress)
```

Comment:
this double branching if/else can probably be cleaned up a little. maybe put the resolution logic into its own function?